src/Microsoft.ML.PCA/PcaTrainer.cs

```diff
@@ -90,20 +90,20 @@ internal static class Defaults
         /// Initializes a new instance of <see cref="RandomizedPcaTrainer"/>.
         /// </summary>
         /// <param name="env">The local instance of the <see cref="IHostEnvironment"/>.</param>
-        /// <param name="features">The name of the feature column.</param>
-        /// <param name="weights">The name of the weight column.</param>
+        /// <param name="featureColumnName">The name of the feature column.</param>
+        /// <param name="exampleWeightColumnName">The name of the weight column.</param>
         /// <param name="rank">The number of components in the PCA.</param>
         /// <param name="oversampling">Oversampling parameter for randomized PCA training.</param>
-        /// <param name="center">If enabled, data is centered to be zero mean.</param>
+        /// <param name="ensureZeroMean">If enabled, data is centered to be zero mean.</param>
         /// <param name="seed">The seed for random number generation.</param>
         internal RandomizedPcaTrainer(IHostEnvironment env,
-            string features,
-            string weights = null,
+            string featureColumnName,
+            string exampleWeightColumnName = null,
```

Comment:
>exampleWeightColumnName [](start = 19, length = 23)

does 'example' need to be in the name? just too long..

Reply:
Yes but it's used everywhere.

---
In reply to: [262785707](https://github.com/dotnet/machinelearning/pull/2865#discussion_r262785707) [](ancestors = 262785707)